kv/kvserver: TestRollbackSyncRangedIntentResolution failed

kv/kvserver.TestRollbackSyncRangedIntentResolution [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3050984&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3050984&tab=artifacts#/) on release-21.1 @ [ac27d325ed08b76408e3bfdec3707c74d5a1c565](https://github.com/cockroachdb/cockroach/commits/ac27d325ed08b76408e3bfdec3707c74d5a1c565):


```
=== RUN   TestRollbackSyncRangedIntentResolution
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRollbackSyncRangedIntentResolution771443750
    test_log_scope.go:74: use -show-logs to present logs inline
    intent_resolver_integration_test.go:214: 
        	Error Trace:	intent_resolver_integration_test.go:214
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestRollbackSyncRangedIntentResolution
    panic.go:617: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRollbackSyncRangedIntentResolution771443750
--- FAIL: TestRollbackSyncRangedIntentResolution (35.37s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestRollbackSyncRangedIntentResolution PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @erikgrinaker
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRollbackSyncRangedIntentResolution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestRollbackSyncRangedIntentResolution [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3073268&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3073268&tab=artifacts#/) on release-21.1 @ [f6956dd58f277823139b5339fc3764309000ff1c](https://github.com/cockroachdb/cockroach/commits/f6956dd58f277823139b5339fc3764309000ff1c):


```
=== RUN   TestRollbackSyncRangedIntentResolution
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRollbackSyncRangedIntentResolution865877681
    test_log_scope.go:74: use -show-logs to present logs inline
    intent_resolver_integration_test.go:214: 
        	Error Trace:	intent_resolver_integration_test.go:214
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestRollbackSyncRangedIntentResolution
    panic.go:617: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRollbackSyncRangedIntentResolution865877681
--- FAIL: TestRollbackSyncRangedIntentResolution (33.81s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestRollbackSyncRangedIntentResolution PKG=./pkg/kv/kvserver TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
/cc @erikgrinaker
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRollbackSyncRangedIntentResolution.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/intent_resolver_integration_test.go
+++ b/pkg/kv/kvserver/intent_resolver_integration_test.go
@@ -186,8 +186,7 @@ func TestContendedIntentWithDependencyCycle(t *testing.T) {
 func TestRollbackSyncRangedIntentResolution(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.UnderRace(t)
-	skip.UnderStress(t)
+	skip.UnderRace(t, "timing-sensitive test")
 
 	ctx := context.Background()
 	srv, _, _ := serverutils.StartServer(t, base.TestServerArgs{
@@ -208,7 +207,7 @@ func TestRollbackSyncRangedIntentResolution(t *testing.T) {
 		batch.Put([]byte(fmt.Sprintf("key%v", i)), []byte("value"))
 	}
 	require.NoError(t, txn.Run(ctx, batch))
-	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
 	require.NoError(t, txn.Rollback(ctx))
 	require.NoError(t, ctx.Err())

